src/test/java/net/masterthought/cucumber/ReportParserTest.java

```diff
@@ -88,4 +92,147 @@ public void parseJsonResults_OnNoExistingFile_ThrowsException() {
 
         reportParser.parseJsonFiles(jsonReports);
     }
+
+    @Test
+    public void parsePropertyFiles_EmptyProperties() {
+
+        // given
+        initWithProperties(EMPTY_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parseClassificationsFiles(propertyFiles);
+
+        // then
+        assertThat(configuration.getClassifications()).isEmpty();
+    }
+
+    @Test
+    public void parsePropertyFiles_Populates_One_File() {
+
+        // given
+        initWithProperties(SAMPLE_ONE_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parseClassificationsFiles(propertyFiles);
+
+        // then
+        assertThat(configuration.getClassifications()).hasSize(5);
+    }
+
+    @Test
+    public void parsePropertyFiles_Populates_Two_Files() {
+
+        // given
+        initWithProperties(SAMPLE_ONE_PROPERTIES, SAMPLE_TWO_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        reportParser.parseClassificationsFiles(propertyFiles);
+
+        // then
+        List<Map.Entry<String, String>> returnedClassifications = configuration.getClassifications();
+        assertThat(returnedClassifications).hasSize(8);
+        assertThat(returnedClassifications).contains(new AbstractMap.SimpleEntry<>("AutUiVersion", "1.25.3"), Index.atIndex(1));
+        assertThat(returnedClassifications).contains(new AbstractMap.SimpleEntry<>("firefoxVersion", "56.0"), Index.atIndex(4));
+        assertThat(returnedClassifications).contains(new AbstractMap.SimpleEntry<>("Proxy", "http=//172.22.240.68:18717"), Index.atIndex(6));
+        assertThat(returnedClassifications).contains(new AbstractMap.SimpleEntry<>("NpmVersion", "5.3.0"), Index.atIndex(7));
+    }
+
+    @Test
+    public void parsePropertyFiles_Populates_Two_Files_One_Empty() {
+
+        // given
+        initWithProperties(SAMPLE_ONE_PROPERTIES);
+        ReportParser reportParser = new ReportParser(configuration);
+        propertyFiles.add("");
+
+        // when
+        reportParser.parseClassificationsFiles(propertyFiles);
+
+        // then
+        List<Map.Entry<String, String>> returnedClassifications = configuration.getClassifications();
+        assertThat(returnedClassifications).hasSize(5);
+        assertThat(returnedClassifications).contains(new AbstractMap.SimpleEntry<>("AutUiVersion", "1.25.3"), Index.atIndex(1));
```

Comment:
would be nice if you can use other mechanism for validating than creating new `AbstractMap` only to satisfy the API. If not possible or hard, I'm OK with this.

Reply:
Your right, however containsExactly doesn't allow me to verify elements at certain indices unfortunately hence use of this. I assume we will keep this so?

Reply:
Leave it as it is, squash and let me merge it!

Reply:
ok i'll have the final (I hope :) ) PR done this evening.